src/sqlfluff/rules/std.py

```diff
@@ -2827,3 +2827,35 @@ def _eval(self, segment, **kwargs):
                         description=("Found USING statement. Expected only ON statements.")
                     )]
         return None
+
+@std_rule_set.register
+class Rule_L033(BaseCrawler):
+    """UNION ALL is preferred over UNION.
+
+    | **Anti-pattern**
+    | In this example, UNION ALL should be preferred over UNION
+
+    .. code-block:: sql
+
+        SELECT a, b FROM table_1 UNION SELECT a, b FROM table_2
+
+    | **Best practice**
+    | Replace UNION with UNION ALL
+
+    .. code-block:: sql
+
+        SELECT a, b FROM table_1 UNION ALL SELECT a, b FROM table_2
+
+    """
+
+    def _eval(self, segment, raw_stack, **kwargs):
+        """Look for UNION keyword not immediately followed by ALL keyword.
+
+        The function does this by checking if there's an ALL keyword between
+        a UNION keyword and a SELECT keyword
+        """
+        if segment.type == 'keyword' and segment.name == 'SELECT':
+            filt_raw_stack = self.filter_meta(raw_stack)
+            if len(filt_raw_stack) > 1 and filt_raw_stack[-2].name == 'UNION':
+                return LintResult(anchor=segment)
```

Comment:
I think here, is there are more _non code_ segments between the `SELECT` and `UNION` keywords I think we might get false alarms here.

An approach you might find useful is it look for a segment of `type` `set_operator`. If you look for `SetOperatorSegment` in `dialects_ansi` you'll see what that kind of structure it will have. If you look for any `set_operator` which has a `UNION` but no `DISTINCT` or `ALL` then we should flag it as a problem.

btw, if you're not familiar with the mysql syntax of `UNION DISTINCT` you can find the detail here: https://dev.mysql.com/doc/refman/5.7/en/union.html#union-distinct-all

I think `UNION DISTINCT` is ok, but a bare `UNION` (i.e. without an `ALL` or `DISTINCT`) should raise a violation.

Reply:
@alanmcruickshank Thanks for the suggestions! I have locally updated the logic to use the set_operator and will be updating the pull request once I've gotten a few clarifications on @NiallRees comment.